metashare/api/views.py

```diff
@@ -221,13 +217,14 @@ def retrieve(self, request, *args, **kwargs):
         # the middle.
         instance = self.get_object()
         conditions = [
-            instance.owner == request.user,
             instance.org_type == SCRATCH_ORG_TYPES.Dev,
             instance.url is not None,
             instance.delete_queued_at is None,
             not instance.currently_capturing_changes,
             not instance.currently_refreshing_changes,
         ]
+        if not request.GET.get("get_unsaved_changes"):
+            conditions.append(instance.owner == request.user)
```

Comment:
@wlonk I did not add tests for these changes because coverage was still at 100%, but feel free to if you like.

Reply:
I'm OK with not changing the tests; we don't have branch coverage set up in Python at this point. We maybe should, but that's another issue.

Also, make the change to `query_params` here too.